### PR TITLE
Update free-programming-books-pt_BR.md with Architecture book

### PR DESCRIPTION
**Pull request by ThiagoBfim**
## What does this PR do?

Add resource(s)

### Description

I added a Book about architecture for PT-BR

### Why is this valuable (or not)?

This is a great book about architecture, with renowned authors, such as Elder Moraes, Karina Varela, Otávio Santana and others.

### How do we know it's really free?

This book is not for sale. This is a public book, created for the community. 

### For book lists, is it a book? For course lists, is it a course? etc.

It's a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by AwesomeArunava** [comment]
your file has an issue with the spacing that the lint says. please remove the blank line from your file and follow the format of the file. After that push the code and your code will be accepted.

File: books/free-programming-books-pt_BR.md [hunk 1]
```diff
@@ -166,6 +166,7 @@
 #### Arquitetura de Software
 
 * [ASP.NET Core architecture](https://docs.microsoft.com/pt-br/dotnet/architecture/modern-web-apps-azure/) (PDF)
+* [Melhores Práticas de Arquitetura de Software na era da Nuvem](https://leanpub.com/manual-arquitetura-software) (Necessário criar uma conta (gratuita) no Leanpub para baixar o livro completo nos formatos PDF, EPUB, MOBI ou pelo próprio site)
```
Comment (Thenlie) [inline books/free-programming-books-pt_BR.md:169]: I don't think this comment after the link is necessary.

**Review by Thenlie — CHANGES_REQUESTED** [review]
Please remove or update the comment on this addition. Once you have done so, update the PR title to include `(changes made)`. Thanks!

File: books/free-programming-books-pt_BR.md (hunk 1 above)
Comment (ThiagoBfim) [reply books/free-programming-books-pt_BR.md]: I agree, I did it just to follow the standard.

**Comment by eshellman** [comment]
sorry for the back and forth, this book needs a Leanpub access note. (The translation for this language is ... wordy, which makes is seem like a commentary, which we don't do. The english version of the access not is `*(Leanpub account or valid email requested)*` but I think consistency within a file is more important than consistency across languages.) 
Improvements to the portuguese access note would be welcomed.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**